test/template.js

```diff
@@ -10,63 +10,107 @@ describe('TemplateGenerator', function () {
-        let result = templateGenerator.getExternalProject("embark-framework/embark");
-        assert.strictEqual(result.url, "https://github.com/embark-framework/embark/archive/master.zip");
+    });
+
+    describe('with git host shortcut', function () {
+
+      it('returns correct info for GitHub shortcut', function () {
+        let result = templateGenerator.getExternalProject("github:embark-framework/embark");
+        assert.strictEqual(result.url, "https://codeload.github.com/embark-framework/embark/tar.gz/master");
+
+        result = templateGenerator.getExternalProject("embark-framework/embark");
+        assert.strictEqual(result.url, "https://codeload.github.com/embark-framework/embark/tar.gz/master");
         assert.strictEqual(result.filePath.replace(/\\/g,'/'), ".embark/templates/embark-framework/embark/master/archive.zip");
+        assert.strictEqual(result.browse, "https://github.com/embark-framework/embark");
+
+        result = templateGenerator.getExternalProject("embark-framework/embark#features/branch");
+        assert.strictEqual(result.url, "https://codeload.github.com/embark-framework/embark/tar.gz/features%2Fbranch");
+        assert.strictEqual(result.filePath.replace(/\\/g,'/'), ".embark/templates/embark-framework/embark/features/branch/archive.zip");
+        assert.strictEqual(result.browse, "https://github.com/embark-framework/embark/tree/features/branch");
       });
 
-      it('return correct zip filename with just username/repo and branch specified', function () {
-        let result = templateGenerator.getExternalProject("embark-framework/embark#develop");
-        assert.strictEqual(result.url, "https://github.com/embark-framework/embark/archive/develop.zip");
-        assert.strictEqual(result.filePath.replace(/\\/g,'/'), ".embark/templates/embark-framework/embark/develop/archive.zip");
+      it('returns correct info for Bitbucket shortcut', function () {
+        let result = templateGenerator.getExternalProject("bitbucket:embark-framework/embark");
+        assert.strictEqual(result.url, "https://bitbucket.org/embark-framework/embark/get/master.tar.gz");
+        assert.strictEqual(result.filePath.replace(/\\/g,'/'), ".embark/templates/embark-framework/embark/master/archive.zip");
+        assert.strictEqual(result.browse, "https://bitbucket.org/embark-framework/embark");
+
+        result = templateGenerator.getExternalProject("bitbucket:embark-framework/embark#features/branch");
+        assert.strictEqual(result.url, "https://bitbucket.org/embark-framework/embark/get/features%2Fbranch.tar.gz");
+        assert.strictEqual(result.filePath.replace(/\\/g,'/'), ".embark/templates/embark-framework/embark/features/branch/archive.zip");
+        assert.strictEqual(result.browse, "https://bitbucket.org/embark-framework/embark/src/features/branch");
       });
 
-      it('return correct zip filename with just embark template specified', function () {
```

Comment:
Are those tests removed on purpose or is this a breaking change?

Reply:
I removed/replaced many of the tests because they were mostly testing the custom string-parser implemented in the bounty PR. The equivalent test in this case is at the top of the suite, within `describe('with named template', function () {...}`.